pkg/controllers/machine/disruption/drift.go

```diff
@@ -100,3 +103,43 @@ func areStaticFieldsDrifted(provisioner *v1alpha5.Provisioner, machine *v1alpha5
 	}
 	return provisionerHash != machineHash
 }
+
+func areNodeRequirementsDrifted(provisioner *v1alpha5.Provisioner, machine *v1alpha5.Machine) bool {
+	return !lo.EveryBy(provisioner.Spec.Requirements, func(provisionerReq v1.NodeSelectorRequirement) bool {
```

Comment:
I think there's a way to build the requirements using `scheduling.NewLabelReqruirements` and comparing compatability rather than having to do this manually.